### ISSUE TITLE
Exam setting: ask for student ID on frontpage; use it as random seed

### ISSUE THREAD

**Issue by christianp** [numbas/Numbas#925]
Motivated by wanting to give each student a stable randomisation, without storing attempt data in the LTI tool.

The student would be asked for their student ID, and then that's used to set the seed for the random number generator before generating the questions.

The marker can then type in the same ID to see the same randomisation.

**Comment by christianp** [comment]
Duplicate of #805

**Comment by christianp** [comment]
Actually, #805 was only about using the student's ID for randomisation, assuming it's given by the SCORM data model. 

This is about asking for the student's name in standalone mode before the exam is generated, which is a bit harder.

The problem at the moment is that the front page, where the student is asked for their name when offline analysis is turned on, is only shown after the questions are generated. We'd have to change the init flow to just load the exam settings, then show the front page, then generate questions only once the "begin" button is clicked. This might take a while, so we'd have to show the loading bar again.
